packages/dds/map/src/index.ts

```diff
@@ -98,3 +98,46 @@ export const SharedMap = {
  * This alias is for legacy compat from when the SharedMap class was exported as public.
  */
 export type SharedMap = ISharedMap;
+
+/**
+ * {@inheritDoc ISharedDirectory}
+ * @sealed
+ * @alpha
+ */
+export const SharedDirectory = {
```

Comment:
This should go in a file other than `index.ts`. Putting it in the same file the factory is defined in seems reasonable (that also has the side-benefit of cutting the circular dep between the DDS class and its factory, if they're defined in different files)

Reply:
I combined the factory and object within the same file, but it appears inevitable that we have to import and export `SharedMap/SharedDirectory` within the same file. I did some tricks such as `import { SharedMap as SharedMapInternal }`, not sure if  there is a better approach. (in the commit https://github.com/microsoft/FluidFramework/pull/20030/commits/0de8bef448bcb105948307199dec33c5db6988d9#diff-8475d479f31499dd12b9b24699ad7b1cc0972e7eee079ad4e0e0281ae6387d1f)